src/react-components/src/components/01-atoms/Text/Headings/Heading.tsx

```diff
@@ -5,38 +5,36 @@ import bem from "../../../../utils/bem";
 export interface HeadingProps {
   level: number;
   id: string;
+  baseClass?: string;
   modifiers?: string[];
   blockName?: string;
 
+  url?: string;
+  urlClass?: string;
+
   text?: string;
   attributes?: {};
 }
 
 export default function Heading(props: React.PropsWithChildren<HeadingProps>) {
-  const { level, id, text, modifiers, blockName, attributes } = props;
+  const { level, id, baseClass, modifiers, url, urlClass, text, blockName, attributes } = props;
 
   if (level < 1 || level > 6) {
     throw new Error("Heading only supports levels 1-6");
   }
 
+  if (!props.children && !text) {
+    throw new Error("Heading has no children, please pass prop: text");
+  }
+
   let content: string | React.ReactNode;
-  if (text) {
-    content = text;
+  if (props.children) {
+    content = url ? React.createElement("a", { href: url, className: urlClass }, props.children) : props.children;
   } else {
-    let isValid: boolean = true;
-    React.Children.forEach(props.children, (child: React.ReactElement) => {
-      if (typeof child !== "string" && child.type !== "b") {
-        isValid = false;
-      }
-    });
-    if (!isValid) {
-      throw new Error("Headings can only be plain text or bold");
-    } else {
-      content = props.children;
-    }
+    content = url ? React.createElement("a", { href: url, className: urlClass }, text) : text;
   }
 
-  let heading_base_class = "heading";
+  let heading_base_class = baseClass ? baseClass : "heading";
```

Comment:
For these type of props, I like using default props. An alternative is to make "heading" a default value when destructuring (line 20):
```
const { level, id, baseClass = "heading", modifiers, ... } = props;
```

Then, `let heading_base_class = baseClass;` will work for "heading" or any passed in value.

Reply:
hmm this won't override baseClass if it exists already?

Reply:
No, it's a default value but if anything is passed to `Heading`, it'll use that for `baseClass` instead.